src/shogun/machine/Machine.cpp

```diff
@@ -33,26 +32,11 @@ Machine::~Machine()
 
 bool Machine::train(std::shared_ptr<Features> data)
 {
-	if (train_require_labels())
-	{
-		if (m_labels == NULL)
-			error("{}@{}: No labels given", get_name(), fmt::ptr(this));
-
-		m_labels->ensure_valid(get_name());
-	}
-
 	auto sub = connect_to_signal_handler();
 	bool result = false;
 
 	if (support_feature_dispatching())
 	{
-		require(data != NULL, "Features not provided!");
-		require(
-		    data->get_num_vectors() == m_labels->get_num_labels(),
-		    "Number of training vectors ({}) does not match number of "
-		    "labels ({})",
-		    data->get_num_vectors(), m_labels->get_num_labels());
-
```

Comment:
these checks are not needed?

Reply:
move these checks to `NonParametricMachine::train`